lib/services/database/comment_repository_service.dart

```diff
@@ -88,27 +89,84 @@ class CommentRepositoryService {
   Future<void> deleteComment(
     PostIdFirestore parentPostId,
     CommentIdFirestore commentId,
-  ) =>
-      _firestore.runTransaction(
-        (transaction) async {
-          final commentRef =
-              _commentsSubCollection(parentPostId).doc(commentId.value);
-
-          final commentDoc = await transaction.get(commentRef);
-
-          if (!commentDoc.exists) {
-            throw Exception("Comment does not exist");
-          }
-
-          transaction.delete(commentRef);
-
-          final postDocRef = _postDocument(parentPostId);
-          transaction.update(
-            postDocRef,
-            {PostData.commentCountField: FieldValue.increment(-1)},
-          );
-        },
+  ) async {
+    final batch = _firestore.batch();
+
+    final commentUpvoteRepository =
+        UpvoteRepositoryService.commentUpvoteRepository(
+      _firestore,
+      parentPostId,
+    );
+
+    await _deleteCommentNoCountUpdate(
+      parentPostId,
+      commentId,
+      batch,
+      commentUpvoteRepository,
+    );
+
+    batch.update(
+      _postDocument(parentPostId),
+      {PostData.commentCountField: FieldValue.increment(-1)},
+    );
+
+    await batch.commit();
+  }
+
+  /// This method will delete all the comments of the post with id [parentPostId].
+  /// Helper method to delete a post. Adds all the deletions to the batch [batch].
+  Future<void> deleteAllComments(
+    PostIdFirestore parentPostId,
+    WriteBatch batch,
+  ) async {
+    final commentsRef = _commentsSubCollection(parentPostId);
+    final comments = await commentsRef.get();
+    final commentUpvoteRepository =
+        UpvoteRepositoryService.commentUpvoteRepository(
+      _firestore,
+      parentPostId,
+    );
+
+    for (final comment in comments.docs) {
+      await _deleteCommentNoCountUpdate(
+        parentPostId,
+        CommentIdFirestore(value: comment.id),
+        batch,
+        commentUpvoteRepository,
+        false,
```

Comment:
I would prefer this to be a named parameter. I think it would improve the code readability in a way that is significant enough for the time taken to change it. What do you think?

Reply:
c31fa834d53b1f3b1c8fb5cef0ffdcebb582c335